### DISCUSSION TITLE
Multiselect field required

### DISCUSSION THREAD

**Discussion by molotow11** [keystonejs/keystone#9008] [Questions] [answered]
I have a problem when migrating a `json` field to a `multiselect`.
For some reason `multiselect` field going to be required in the db and I see the following message:

```
- Made the column `colors` on table `Test` required, but there are 2 existing NULL values
To apply this migration, we need to reset the database
```

I've tried to add `default: []` to the multiselect field definition, but no luck. Any other properties not available for this field. 
Any ideas how to solve it?

**Comment by dcousens** [comment, marked as answer]
Your data is already in the database, and the database column type is probably unchanged when moving from `json` to `multiselect`, except in that it has now become `NOT NULL`.

Default values do not typically apply except for new data from an SQL `INSERT`, thereby, you need to migrate this yourself.

**Reply by molotow11** [reply]
Thanks! How it is going to be `NOT NULL` ? How it is possible to fix it?

**Reply by dcousens** [reply]
The only way to resolve this is to migrate your data, as the two types are not compatible.
